script_based_workflows.Rmd

```diff
@@ -0,0 +1,201 @@
+
+`code_to_function("01_data.R")` which generates the function `load_data()` and is not dependent on any other steps, so it has no targets passed into it and generates the `data` target."02_munge.R" is dependent on "01_data.R", and `drake` is informed by passing the target `data` as an argument to `do_munge()`. The same goes for "03_histogram.R", with `munged_data` as an argument for `generate_histogram()`.
+
+```{r}
+
+simple_plan <- drake_plan(
+  data     = load_data(),
+  munged_data = do_munge(data),
+  hist     = generate_histogram(munged_data)
+)
+```
+
+```{r echo=FALSE}
+
+simple_config <- drake_config(simple_plan)
+vis_drake_graph(simple_config)
+
+```
+
+
+## Putting it all together
+
+Using the first example of a make script in this chapter as the framework, the new `make.R` would look as follows.
+
+```{r eval=FALSE}
+
+# convert the scrips to functions
+source("functions.R")
+
+# Create Drake Plan
+script_based_plan <- drake_plan(
+  data        = load_data(),
+  munged_data = do_munge(data),
+  hist        = do_histogram(munged_data),
+  fit         = do_regression(munged_data),
+  report      = generate_report(hist, fit)
+)
+
+make(script_based_plan)
+
+```
+
+where `functions.R` contains the code to convert the scripts to functions:
+
+```{r eval=FALSE}
+# Convert Script based workflow to functions
+load_data <- code_to_function("01_data.R")
+do_munge <- code_to_function("02_munge.R")
+do_histogram <- code_to_function("03_histogram.R")
+do_regression <- code_to_function("04_regression.R")
+generate_report <- code_to_function("05_report.R")
+
+```
+
+By converting everything as a function and passing targets of the prior dependencies, `drake` is able track dependencies and when make.R is rerun, only the updated scripts will be rerun.
+
+```{r echo=FALSE}
+# Convert Script based workflow to functions
+load_data <- code_to_function("01_data.R")
+do_munge <- code_to_function("02_munge.R")
+do_histogram <- code_to_function("03_histogram.R")
+do_regression <- code_to_function("04_regression.R")
+generate_report <- code_to_function("05_report.R")
+
+# Create Drake Plan
+script_based_plan <- drake_plan(
+  data        = load_data(),
+  munged_data = do_munge(data),
+  hist        = do_histogram(munged_data),
+  fit         = do_regression(munged_data),
+  report      = generate_report(hist, fit)
+)
+
+```
+
+```{r}
+
+script_based_config <- drake_config(script_based_plan)
+vis_drake_graph(script_based_config, targets_only = TRUE)
+```
+
+
+Any time a script is updated, either `functions.R` can be sourced again to update the representative function, or the specific `code_to_function()` line can be rerun. `drake` can see that the function is updated and when `make()` is called, it will rerun the updated function and all dependencies. 
```

Comment:
I think a demo of this, plus a couple `vis_drake_graph()`s, may be helpful.

Reply:
Done :)